### PR TITLE
Explain to load different configs depending on the domain in `config.md`

### PR DESCRIPTION
**Pull request by dklimpel**
Add docs for
- #4575

## Checklist

* [x] Tests written for new code (and old code if feasible)
* [x] Linter and other CI checks pass
* [x] Sign-off given on the changes (see [CONTRIBUTING.md](https://github.com/vector-im/element-web/blob/develop/CONTRIBUTING.md))

Signed-off-by: Dirk Klimpel dirk@klimpel.org

---
This change is marked as an *internal change* (Task), so will not be included in the changelog.

### CODE REVIEW

File: docs/config.md [hunk 1]
```diff
@@ -1,8 +1,13 @@
 # Configuration
 
-You can configure the app by copying `config.sample.json` to `config.json` and customising it. The possible options are
-described here. If you run into issues, please visit [#element-web:matrix.org](https://matrix.to/#/#element-web:matrix.org)
-on Matrix.
+You can configure the app by copying `config.sample.json` to `config.json` or `config.$domain.json` and customising it.
+Element will attempt to load first `config.$domain.json` and if it fails `config.json`. This mechanism allows different
+configuration options depending on if you're hitting e.g. `app1.example.com` or `app2.example.com`. Configs are not mixed
+in any way, it either entirely uses the domain config, or entirely uses `config.json`. Using `config.json` is the
+recommended default solution.
```
Comment (SimonBrandner) [inline docs/config.md:7]: What leads you to think it's recommended?

**Review by SimonBrandner — COMMENTED** [review]
Thanks for working on this! Could you please include a sign-off as described [here](https://github.com/matrix-org/matrix-js-sdk/blob/develop/CONTRIBUTING.md#sign-off)?

File: docs/config.md (hunk 1 above)
Comment (dklimpel) [reply docs/config.md:7]: I don't have a strong preference to this.
The thought was that if it wasn't, an alternative would have been documented over the last 4 years.
Public element.io seems to use it that way too.

Comment (SimonBrandner) [reply docs/config.md:7]: I would agree that it arguably is the "standard" way to do this though I wouldn't go as far as to say it is recommended. My suggestion here would be just to remove the sentence as it feels unnecessary

**Comment by dklimpel** [comment]
> Thanks for working on this! Could you please include a sign-off as described [here](https://github.com/matrix-org/matrix-js-sdk/blob/develop/CONTRIBUTING.md#sign-off)?

I am not sure what is wrong with the sign off?
> If you agree to this for your contribution, then all that's needed is to include the line in your commit or pull request comment

This is exactly the same wording as https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#sign-off
and I did the same. I did the sing off in PR comment, as described.

File: docs/config.md (hunk 1 above)
Comment (dklimpel) [reply docs/config.md]: I have removed recommendation.

**Review by SimonBrandner — APPROVED** [review]
Sorry, I missed the sign-off!

**Pull request merged**